helm-charts/splunk-otel-collector/scripts/lookup-eks-fargate-receiver-node.sh

```diff
@@ -0,0 +1,51 @@
+#! /usr/bin/bash
+set -ex
+
+# If we are the first pod (cluster receiver), set the kubelet stats node filter to only follow labelled nodes.
+# This node label will be set by the second pod.
+if [[ "${K8S_POD_NAME}" == *-0 ]]; then
+  echo "will configure kubelet stats receiver to follow node ${FIRST_CR_REPLICA_NODE_NAME}, as well as use cluster receiver."
+  echo "export CR_KUBELET_STATS_NODE_FILTER='&& labels[\"splunk-otel-is-eks-fargate-cluster-receiver-node\"] == \"true\"'" >/splunk-messages/environ
+  echo "export CR_K8S_OBSERVER_OBSERVE_PODS='false'" >>/splunk-messages/environ
+
+  cat /splunk-messages/environ
+
+  # copy config to meet container command args
+  cp /conf/relay.yaml /splunk-messages/config.yaml
+  exit 0
+fi
+
+# Else we are the second pod (wide kubelet stats) label our node to be monitored by the first pod and disable the k8s_cluster receiver.
+# Update our config to not monitor ourselves
+echo "Labelling our fargate node to denote it hosts the cluster receiver"
+
+# download kubectl (verifying checksum)
+curl -o kubectl https://amazon-eks.s3.us-west-2.amazonaws.com/1.16.15/2020-11-02/bin/linux/amd64/kubectl
```

Comment:
Why do we use 1.16 version? It's not supported anymore I believe

Reply:
This is the oldest provided version and fargate nodes were supported on 1.16 clusters. I'm not well versed in kubectl compatibility, so what version would you suggest?

Reply:
Kubernetes client and server minor version skew is +/-1. It doesn't mean difference beyond the skew cannot work, but it's not guaranteed to work.

Probably we can use built-in template objects to get current user kubernetes version https://helm.sh/docs/chart_template_guide/builtin_objects/ but in that case hardcoded checksum worn work :)

Reply:
From the other hand it's just one operation. So if we can test it manually, it should be ok to use a version somewhere in between 1.16 and the latest kubernetes version 1.23, maybe 1.20

Reply:
1.20 it is